Mathlib/RingTheory/HahnSeries/Summable.lean

```diff
@@ -0,0 +1,592 @@
+/-
+Copyright (c) 2021 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Anderson
+-/
+import Mathlib.RingTheory.HahnSeries.Multiplication
+import Mathlib.Algebra.EuclideanDomain.Instances
+import Mathlib.Algebra.Order.Group.WithTop
+import Mathlib.RingTheory.Valuation.Basic
+
+#align_import ring_theory.hahn_series from "leanprover-community/mathlib"@"a484a7d0eade4e1268f4fb402859b6686037f965"
+
+/-!
+# Hahn Series
+If `Γ` is ordered and `R` has zero, then `HahnSeries Γ R` consists of formal series over `Γ` with
+coefficients in `R`, whose supports are partially well-ordered. With further structure on `R` and
+`Γ`, we can add further structure on `HahnSeries Γ R`.  We introduce valuations and a notion of
+summability for possibly infinite families of series.
+## Main Definitions
```

Comment:
Minor thing, but I think an empty line before subheaders makes the docstring a bit more readable.
```suggestion

## Main Definitions
```